src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -0,0 +1,298 @@
+                    output.get_tensor_ptr()->set_names({});
+                }
+            }
+        }
+    }
+
+    for (auto it = result_types.begin(); it != result_types.end(); ++it) {
+        const auto result = it->first;
+        const auto actual_type = result->get_input_element_type(0);
+        const auto expected_type = it->second;
+        if (actual_type != it->second) {
+            was_updated = true;
+            auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
+                result->get_input_node_shared_ptr(0),
+                expected_type);
+            ngraph::copy_runtime_info(result->get_input_node_shared_ptr(0), convert);
+            result->set_argument(0, convert);
+        }
+    }
+
+    return was_updated;
+}
+
+bool ngraph::snippets::pass::PropagatePrecision::can_be_removed(
+    const element::Type& actual_before,
+    const element::Type& actual_after,
+    const element::Type& required_after) noexcept {
+    if (actual_before != required_after) {
+        return false;
+    }
+
+    // CPU support: avoid FP32 => BF16 => FP32
+    if ((actual_before == element::f32) && (actual_after == element::bf16)) {
+        return true;
+    }
+
+    return can_be_fused(actual_after, actual_before);
+}
+
+bool ngraph::snippets::pass::PropagatePrecision::can_be_fused(
+    const element::Type& actual,
+    const element::Type& required) noexcept {
+    // custom conditions: between int & float precisions
+    if (((actual == element::bf16) || (actual == element::f16) || (actual == element::f32)) &&
+        ((required == element::u8) || (required == element::i8))) {
+        return true;
+    }
+
+    if ((actual == element::f32) && ((required == element::u16) || (required == element::i16))) {
+        return true;
+    }
+
+    // general conditions: any new added precision will support
+    return
+        (actual.is_real() == required.is_real()) &&
+        (actual.bitwidth() >= required.bitwidth());
+}
+
+std::vector<element::Type> ngraph::snippets::pass::PropagatePrecision::get_precisions(
+    const std::vector<element::Type>& input_precisions,
+    const std::set<std::vector<element::Type>>& supported_precisions_pack) noexcept {
+    bool was_found = false;
+    for (const auto& supported_precisions : supported_precisions_pack) {
+        for (size_t i = 0; i < supported_precisions.size(); ++i) {
+            const auto& supported_precision = supported_precisions[i];
+            const auto& input_precision = input_precisions[i];
+            if ((supported_precision.is_real() != input_precision.is_real()) ||
+                (input_precision.bitwidth() > supported_precision.bitwidth())) {
+                was_found = false;
+                break;
+            }
+
+            was_found = true;
+        }
+        if (was_found) {
+            return supported_precisions;
+        }
+    }
+
+    if (!supported_precisions_pack.empty()) {
+        return *supported_precisions_pack.begin();
```

Comment:
Minor
```suggestion
        return supported_precisions_pack.front();
```

Reply:
`supported_precisions_pack` is `std::set`, there is no `front` here